src/frontend/src/views/Upload/components/Metadata/index.tsx

```diff
@@ -33,24 +33,57 @@ import {
   ParseResult,
   SampleIdToWarningMessages,
 } from "./components/ImportFile/parseFile";
+import { analyticsTrackEvent } from "src/common/analytics/methods";
+import {
+  AnalyticsUploadMetadataType,
+  EVENT_TYPES,
+} from "src/common/analytics/eventTypes";
+import { useSelector } from "react-redux";
+import { selectCurrentPathogen } from "src/common/redux/selectors";
 
 export default function Metadata({
   samples,
   namedLocations,
   metadata,
   setMetadata,
+  analyticsFlowUuid,
 }: Props): JSX.Element {
+  const pathogen = useSelector(selectCurrentPathogen);
   const [isValid, setIsValid] = useState(false);
   const [hasImportedMetadataFile, setHasImportedMetadataFile] =
     useState<boolean>(false);
   const [autocorrectWarnings, setAutocorrectWarnings] =
     useState<SampleIdToWarningMessages>(EMPTY_OBJECT);
+  // this is used to track whether the user has manually edited the metadata for analytics
+  const [hasManuallyEditedMetadata, setHasManuallyEditedMetadata] =
+    useState<boolean>(false);
 
   let numberOfDetectedSamples = 0;
   if (samples != null) {
-    numberOfDetectedSamples = Object.keys(samples).length;
+    numberOfDetectedSamples = (Object.keys(samples).length);
```

Comment:
what's up with these parens?

Reply:
oh woops that was an artifact from something that i was trying earlier, thanks for catching